realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java

```diff
@@ -0,0 +1,379 @@
+            String methodName = executableElement.getSimpleName().toString();
+
+            // Check the modifiers of the method
+            Set<Modifier> modifiers = executableElement.getModifiers();
+            if (modifiers.contains(Modifier.STATIC)) {
+                continue; // We're cool with static methods. Move along!
+            } else if (!modifiers.contains(Modifier.PUBLIC)) {
+                Utils.error("The methods of the model must be public", executableElement);
+            }
+
+            if (methodName.startsWith("get") || methodName.startsWith("is")) {
+                checkGetterMethod(methodName);
+            } else if (methodName.startsWith("set")) {
+                checkSetterMethod(methodName);
+            } else {
+                Utils.error("Only getters and setters should be defined in model classes", executableElement);
+            }
+        }
+    }
+
+    // Verify that a setter is used to set a field in the model class.
+    // Note: This is done heuristically by comparing the name of setter with the name of the field.
+    // Annotation processors does not allow us to inspect individual statements.
+    private void checkSetterMethod(String methodName) {
+        boolean found = false;
+
+        String methodMinusSet = methodName.substring(3);
+        String methodMinusSetCapitalised = Utils.lowerFirstChar(methodMinusSet);
+        String methodMenusSetPlusIs = "is" + methodMinusSet;
+
+        if (fieldNames.contains(methodMinusSet)) { // mPerson -> setmPerson
+            expectedSetters.remove(methodMinusSet);
+            if (!ignoreFieldNames.contains(methodMinusSet)) {
+                setters.put(methodMinusSet, methodName);
+            }
+            found = true;
+        } else if (fieldNames.contains(methodMinusSetCapitalised)) { // person -> setPerson
+            expectedSetters.remove(methodMinusSetCapitalised);
+            if (!ignoreFieldNames.contains(methodMinusSetCapitalised)) {
+                setters.put(methodMinusSetCapitalised, methodName);
+            }
+            found = true;
+        } else if (fieldNames.contains(methodMenusSetPlusIs)) { // isReady -> setReady
+            expectedSetters.remove(methodMenusSetPlusIs);
+            if (!ignoreFieldNames.contains(methodMenusSetPlusIs)) {
+                setters.put(methodMenusSetPlusIs, methodName);
+            }
+            found = true;
+        }
+
+        if (!found) {
+            Utils.note(String.format("Setter %s is not associated to any field", methodName));
+        }
+    }
+
+    // Verify that a getter is used to get a field in the model class.
+    // Note: This is done heuristically by comparing the name of getter with the name of the field.
+    // Annotation processors does not allow us to inspect individual statements.
+    private void checkGetterMethod(String methodName) {
+        boolean found = false;
+
+        if (methodName.startsWith("is")) {
```

Comment:
These methods should only exist for boolean fields. Isn't this the right place to check? I mean, the method is called `checkGetterMethod` but doesn't check the type.